doc/custom_nonconformity_score_proof.ipynb

```diff
@@ -0,0 +1,407 @@
+    "Let derive the proof of the theorem to build [the jackknife+ prediction interval](https://arxiv.org/pdf/1905.02928.pdf) (Theorem 1, part 6, p. 18-21) with more general nonconformity scores.\n",
+    "\n",
+    "## R contruction\n",
+    "Instead of defining the R matrix with the absolute residual, let assume that R is defined by means of a nonconformity score $f$:\n",
+    "$$\\begin{align}\n",
+    "R_{i,j} &= f(Y_{i}, \\hat{\\mu}_{-(i,j)}(X_{i})) \\\\\n",
+    "R_{i}^{LOO} &= R_{i,n+1}\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## A contruction\n",
+    "There are two variants:\n",
+    "- the symmetrical case when the conformity scores are assumed to be symmetrical around 0. In this case, the absolute values of the conformity scores are used to construct the A matrix.\n",
+    "- otherwise, the signed conformity scores are kept as they are.\n",
+    "\n",
+    "### Symmetrical case\n",
+    "Like in the paper, the A matrix is defined as below:\n",
+    "$$A_{i,j} = \\mathbb{1} \\{ \\lvert R_{i,j} \\rvert > \\lvert R_{j,i} \\rvert \\}$$\n",
+    "\n",
+    "### Asymmetrical case\n",
+    "The two following definitions of matrix A are considered: \n",
+    "$$\\begin{align}\n",
+    "A^{+}_{i,j} &= \\mathbb{1} \\{R_{i,j} > R_{j,i}\\} \\\\\n",
+    "A^{-}_{i,j} &= \\mathbb{1} \\{R_{i,j} < R_{j,i}\\} = J - A^{+}\n",
+    "\\end{align}$$\n",
+    "\n",
+    "## Strange points\n",
+    "Like previously, the definition of strange points depends on the case considered.\n",
+    "\n",
+    "### Symmetrical case\n",
+    "Like in the paper, there is a single set $S(A)=\\{i \\in \\{1,...,n+1\\}:A_{i.}\\geq(1-\\alpha)(n+1)\\}$.\n",
+    "\n",
+    "### Asymmetrical case\n",
+    "Instead of the single $S(A)$ set, two cases should be considered: $A^{+}$ and $A^{-}$. Keeping the same definition of the set $S$ as the paper, we have:\n",
+    "- $S_{+}=S(A^{+})=\\{i \\in \\{1,...,n+1\\}:A^{+}_{i.}\\geq(1-\\frac{\\alpha}{2})(n+1)\\}$ for \"data points with unusually large [positive] residuals\", and\n",
+    "- $S_{-}=S(A^{-})=\\{i \\in \\{1,...,n+1\\}:A^{-}_{i.}\\geq(1-\\frac{\\alpha}{2})(n+1) \\iff (n-A^{+}_{i.})\\geq(1-\\frac{\\alpha}{2})(n+1)\\}$  for \"data points with unusually large [negative] residuals\".\n",
+    "\n",
+    "## Step 1 and 2 are unchanged\n",
+    "Changing $R$ and $A$, does not change the results of the steps 1 and 2. From step 1, we have that, in the symetrical case $|S(A)| \\leq 2 \\alpha (n+1)$. And in the asymmetrical case $|S_{+}| \\leq \\alpha (n+1)$ and $|S_{-}| \\leq \\alpha (n+1)$ since the new matrices have the same property as from the paper. From step 2, we have that the probability of the test point $n+1$ being strange inside $S_{+}$ is bounded by $\\alpha$ and the probability of the test point $n+1$ being strange inside $S_{-}$ is also bounded by $\\alpha$ for the asymmetrical case (and the probability of the test point $n+1$ being strange inside $S(A)$ is bounded by $2\\alpha$ for the symmetrical case).\n",
```

Comment:
of the steps 1 and 2 -> of steps 1 and 2
we have that, in the symetrical case $|S(A)| -> in the symetrical case, we get $|S(A)|
matrices -> arrays 
as from the paper -> as in the paper ?

Reply:
- Done
- Done
- Same comment as previously.
- Done